src/core/config/ping_sender.h

```diff
@@ -57,6 +57,17 @@
 #define OPENTHREAD_CONFIG_PING_SENDER_DEFAULT_INTEVRAL 1000
 #endif
 
+/**
+ * @def OPENTHREAD_CONFIG_PING_SENDER_DEFAULT_DEADLINE
+ *
+ * Specifies the default ping timeout in milliseconds. The timeout is the max acceptable time gap between each ping
+ * request and its reply.
+ *
+ */
+#ifndef OPENTHREAD_CONFIG_PING_SENDER_DEFAULT_TIMEOUT
+#define OPENTHREAD_CONFIG_PING_SENDER_DEFAULT_TIMEOUT 2000
```

Comment:
```suggestion
#define OPENTHREAD_CONFIG_PING_SENDER_DEFAULT_TIMEOUT 3000
```
nit: 2000 is quite small.  Can 3000 work?